Group-04/lib/IndexSet.hpp

```
@@ -280,10 +395,14 @@ class IndexSet {
    * @return true if the index is in the set, false otherwise
```

(commenting here since I can't comment above on line 341). You could consider using cse_assert to validate valid input instead of just having a silent error you could test for it. Something like:
cse_assert(start < end && "IndexSet::insert_range called with invalid range");

---

Fixed in ac89a95